internal/report/report.go

```diff
@@ -620,24 +621,22 @@ func printTags(w io.Writer, rpt *Report) error {
 	for _, s := range p.Sample {
 		for key, vals := range s.Label {
 			for _, val := range vals {
-				if valueMap, ok := tagMap[key]; ok {
-					valueMap[val] = valueMap[val] + s.Value[0]
-					continue
+				valueMap, ok := tagMap[key]
+				if !ok {
+					valueMap = make(map[string]int64)
 				}
-				valueMap := make(map[string]int64)
-				valueMap[val] = s.Value[0]
+				valueMap[val] += o.SampleValue(s.Value)
 				tagMap[key] = valueMap
```

Comment:
throughout: maps are references, so I believe this can move inside the `if !ok {` scope